website/docs/r/stack_hci_logical_network.html.markdown

```diff
@@ -0,0 +1,123 @@
+    address_prefix       = "10.0.0.0/24"
+    ip_pool {
+      start = "10.0.0.218"
+      end   = "10.0.0.230"
+    }
+    route {
+      name                = "example-route"
+      address_prefix      = "10.0.0.0/28"
+      next_hop_ip_address = "10.0.20.1"
+    }
+    vlan_id = 123
+  }
+
+  tags = {
+    foo = "bar"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Azure Stack HCI Logical Network. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.
+
+* `custom_location_id` - (Required) The ID of Custom Location where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.
+
+* `vm_switch_name` - (Required) The name of the network switch used to associate with the Azure Stack HCI Logical Network. Possible switch name can be retrieved following the [Azure documents](https://learn.microsoft.com/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites). Changing this forces a new resource to be created.
+
+* `subnet` - (Required) A `subnet` block as defined below. Changing this forces a new resource to be created.
+
+* `dns_servers` - (Optional) A list of IPv4 addresses of DNS servers available to VMs deployed in the Logical Networks. Changing this forces a new resource to be created.
+
+* `tags` - (Optional) A mapping of tags which should be assigned to the Azure Stack HCI Logical Network.
+
+---
+
+A `ip_pool` block supports the following:
+
+* `start` - (Required) The IPv4 address of the start of the IP address pool. Changing this forces a new resource to be created.
+
+* `end` - (Required) The IPv4 address of the end of the IP address pool. Changing this forces a new resource to be created.
+
+---
+
+A `route` block supports the following:
+
+* `name` - (Required) The name of the route. Changing this forces a new resource to be created.
+
+* `address_prefix` - (Optional) The Address in CIDR notation. Changing this forces a new resource to be created.
+
+* `next_hop_ip_address` - (Optional) The IPv4 address of the next hop. Changing this forces a new resource to be created.
+
+---
+
+A `subnet` block supports the following:
+
+* `ip_allocation_method` - (Required) IP address allocation method for the subnet. Possible value is `Dynamic` or `Static`. Changing this forces a new resource to be created.
+
+* `address_prefix` - (Optional) The address prefix in CIDR notation. Changing this forces a new resource to be created.
+
+* `ip_pool` - (Optional) One or more `ip_pool` block as defined above. Changing this forces a new resource to be created.
```

Comment:
why not define the description of `ip_pool` and `route` after `subnet` ?

Reply:
all the blocks are ranked in alphabetical order